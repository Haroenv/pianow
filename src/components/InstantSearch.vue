<template lang="pug">
  .instant-search
    slot
      h1 You need to put a search-index element here :)
</template>

<script>
import algoliasearch from "algoliasearch";

export default {
  name: "InstantSearch",
  props: {
    appId: {
      type: String,
      required: true
    },
    apiKey: {
      type: String,
      required: true
    },
    options: {
      type: Object,
      default: null
    }
  },
  provide() {
    return {
      _client: algoliasearch(this.appId, this.apiKey, Object.assign({}, this.options))
    };
  }
};
</script>
