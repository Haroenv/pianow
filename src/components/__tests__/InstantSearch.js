import Vue from "vue";
import InstantSearch from "../InstantSearch";
import algoliasearch from "algoliasearch";
jest.mock("algoliasearch");

describe("InstantSearch.vue", () => {
  let Constructor;
  let vm;
  const options = {
    hasProperty: true
  }

  beforeEach(() => {
    algoliasearch.mockClear();
  });

  it("has a provide function which return an object with a _client property", () => {
    expect(typeof InstantSearch.provide).toBe("function");
    const provide = InstantSearch.provide();
    expect(typeof provide).toBe("object");
    expect(provide.hasOwnProperty('_client')).toBeTruthy();
  });

  beforeEach(() => {
    Constructor = Vue.extend(InstantSearch);
    vm = new Constructor({
      propsData: {
        appId: "appId",
        apiKey: "apiKey",
        options: options
      }
    }).$mount();
  });

  it("should call algoliasearch client with correct component props", () => {
    expect(algoliasearch).toHaveBeenCalledTimes(1);
    expect(algoliasearch).toBeCalledWith("appId", "apiKey", options);
  });

  it("should render correct contents", () => {
    expect(vm.$el.querySelector(".instant-search h1").textContent).toEqual(
      "You need to put a search-index element here :)"
    );
  });

  it("should match the snapshot", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
