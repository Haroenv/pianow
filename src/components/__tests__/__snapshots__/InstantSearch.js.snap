// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstantSearch.vue should match the snapshot 1`] = `
<div
  class="instant-search"
>
  <h1>
    You need to put a search-index element here :)
  </h1>
</div>
`;
